monero/wallet_client.go

```diff
@@ -116,33 +105,72 @@ func NewWalletClient(conf *WalletClientConf) (WalletClient, error) {
 	}
 	isNewWallet := !walletExists
 
-	proc, monerodNode, err := createWalletRPCService(conf)
+	if conf.MoneroWalletRPCPath == "" {
+		conf.MoneroWalletRPCPath, err = getMoneroWalletRPCBin()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if len(conf.MonerodNodes) == 0 {
+		conf.MonerodNodes = common.ConfigDefaultsForEnv(conf.Env).MoneroNodes
+	}
+	validatedNode, err := findWorkingNode(conf.Env, conf.MonerodNodes)
```

Comment:
I moved the primary-wallet validation code (above and below) out of `createWalletRPCService` and into this `NewWalletClient` function. The configurations for the A+B wallets are generated by `CreateABWalletConf` and don't need to be re-validated.